src/views/CreateView.vue

```diff
@@ -208,237 +209,221 @@
   </main>
 </template>
 
-<script>
-import { mapState } from 'pinia'
+<script setup lang="ts">
+import { ref, Ref, computed, onMounted } from 'vue'
+import { useRouter } from 'vue-router'
+import { useUserStore } from '../stores/user'
+import { handleModal } from '../composables/handleModal'
+import { supabase } from '../supabase'
+import { Exercise, ExerciseResponse, UserResponse } from '../api/types'
 import TheForm from '../components/shared/TheForm.vue'
 import TheInput from '../components/shared/TheInput.vue'
 import TheButton from '../components/shared/TheButton.vue'
 import TheExercise from '../components/shared/TheExercise.vue'
 import NeedSignIn from '../components/shared/NeedSignIn.vue'
 import TheToast from '../components/shared/TheToast.vue'
-import { useUserStore } from '../stores/user'
-import { supabase } from '../supabase'
 
-export default {
-  name: 'CreateView',
-  components: {
-    TheForm,
-    TheInput,
-    TheButton,
-    TheExercise,
-    NeedSignIn,
-    TheToast
-  },
-  data() {
-    return {
-      exercises: [],
-      workoutName: '',
-      exerciseName: '',
-      isNameShown: false,
-      areExercisesDisplayed: false,
-      selectedExercise: '',
-      kgValue: false,
-      lbValue: false,
-      exerciseData: {
-        name: '',
-        img: '',
-        muscles: '',
-        sets: '',
-        reps: '',
-        weight: '',
-        unit: '',
-        rest: ''
-      },
-      addedExercises: [],
-      isExerciseSubmitted: false,
-      isToastShown: false,
-      isInsertionError: false
+const exercises = ref<Exercise[]>([])
+const workoutName = ref('')
+const exerciseName = ref('')
+let selectedExercise: Ref<Exercise> | Ref<null> = ref(null)
```

Comment:
const

Reply:
**Corrected**

Reply:
This is not fixed yet. Using `ref()` creates a reactive object. So we get and set its value using `.value` property. There are actually 4 mistakes:
- incorrect typing
- don't set objects to `let`, as you can still set it's property when it's const.
- remove logic from template into separate handlers
- keep appropriate order for event handlers (dunno why it's not linted)


## At first let's fix the types:
```ts
const selectedExercise: Ref<Exercise | null> = ref(null)
```

## Second and the third one 
are the reasons you've decided to use `let` I believe.

If you take a look at you `@click` handler now, it does two things (which is not true, as it always works the same).
```jsx
<li @click.capture="selectedExercise = exercise; addExerciseInfo()">
```
 you set `selectedExercise` to `exercise` - what actually happens there is you set it to a `ref` so it's a vue reactive property. As you use it in the template you should know that using reactive property in the template access its `.value` under the hood. If I have a `const foo = ref("Hello")` and I want to use it in the template I'd write `{{ foo }}`, not `{{ foo.value }}` as `.value` but it's `foo.value` which is actually accessed.

So you should change this let to `const`, that will result a proper Typescript error, but we'll get to that soon. Now let's follow the code that is executed in that `@click`.  After a value is set to `selectedExercise` you call `addExerciseInfo()`. That function is declared in the component's script and triggered only once, which is fine, but take a look what happens in it's body:

```ts
const addExerciseInfo = () => {
  if (selectedExercise.value != null) {
    exerciseData.value.name = selectedExercise.value.name
    exerciseData.value.img = selectedExercise.value.img
    exerciseData.value.muscles = selectedExercise.value.muscles
  }
}
```

My point is we could pass the exercise as a function parameter, set it as `selectedExercise` and then assign the values to the `exerciseData`. It would look like this:

```jsx
<li @click.capture="() => setExercise(exercise)">
```

```ts
const setExercise = (exercise: Exercise) => {
    selectedExercise.value = exercise;
    exerciseData.value.name = exercise.name
    exerciseData.value.img = exercise.img
    exerciseData.value.muscles = exercise.muscles
}
```

## Fourth

I've realised your eslint config is incorrect. I'll fix it for you in a separate PR

Reply:
Clever 🙃
I wouldn't think like that (yet?) but I'll try to remember for the future.

**Corrected**